src/Microsoft/Provider.php

```diff
@@ -149,7 +150,7 @@ protected function getUserByToken($token)
                     ]
                 );
                 $formattedResponse['tenant'] = json_decode((string) $responseTenant->getBody(), true)['value'][0] ?? null;
-            } catch (ClientException) {
+            } catch (ServerException) {
```

Comment:
Should we not catch both here? Or will it only ever return a 500 in the case the tenant does not exist?

Reply:
I am not entirely sure because Microsoft doesn't have great documentation on what specific HTTP error codes are produced and when. From my experience, it as only ever returned 500 when the tenant does not exists. It could be related to this PR [https://github.com/SocialiteProviders/Providers/pull/1035](https://github.com/SocialiteProviders/Providers/pull/1035/)

I actually like your idea of catching both exceptions. My theory is that the `ClientException` occurs when the the token you use the fetch the tenant is not allowed to access information from that tenant (caused by trying to get a tenant using a token from a user who is not a member of said tenant). The `ServerException` happens when the user has no tenant (ex. a consumer account) or the tenant does not exists at all. Either way I can't find any documentation about it so I have no way to confirm without testing.

Reply:
Maybe just catch `RequestException` which will cover both cases?

Reply:
Sounds good, I will go ahead and add that now!